### PR TITLE
[plugin.video.funcarrun] 0.0.3

### PR DESCRIPTION
**Pull request by Paul-VW**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.funcarrun/resources/lib/plugin.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        videoId = v['videoId']
+        uri = "plugin://plugin.video.youtube/?action=play_video&videoid=" + videoId
+        title = v['title']
+        description = v['description']
+        thumbnail = v['thumbUrl']
+        fanart = v['fanartUrl']
+        description = description + "\n\nKijk voor inschrijven op https://www.funcarrun.eu"
+        listitem = xbmcgui.ListItem(label=title)
+        listitem.setInfo('video', {'plot': description})
+        listitem.setInfo('video', {'plotoutline': description})
```
Comment (enen92) [inline plugin.video.funcarrun/resources/lib/plugin.py]: You can pass both infotags properties in a single call (it's a dict)

File: plugin.video.funcarrun/resources/lib/plugin.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        listitem.setProperty('IsPlayable', 'true')
+        listitem.setArt({'icon': thumbnail, 'poster': thumbnail})
+        listitem.setProperty('fanart_image', fanart)
```
Comment (enen92) [inline plugin.video.funcarrun/resources/lib/plugin.py]: Remove this property, no longer used

File: plugin.video.funcarrun/resources/lib/plugin.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+def list_categories():
+    url = "https://www.funcarrun.eu/apiv2.php?type=video_categories"
+    r = requests.get(url)
+    result = r.text
+    categories = json.loads(result)
+
+    listing = []
+    for category in categories:
+        list_item = xbmcgui.ListItem(label=category['title'])
+        list_item.setProperty('fanart_image', category['fanart'])
```
Comment (enen92) [inline plugin.video.funcarrun/resources/lib/plugin.py]: Same

File: plugin.video.funcarrun/resources/lib/plugin.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        videos_by_category(category=category)
+    else:
+        list_categories()
+
+    xbmcplugin.endOfDirectory(__handle__)
+
+
+def start():
+    videoList = get_channel_content()
+    print(videoList)
```
Comment (enen92) [inline plugin.video.funcarrun/resources/lib/plugin.py]: No prints please, use xbmc.log()

**Review by enen92 — CHANGES_REQUESTED** [review]

File: plugin.video.funcarrun/resources/lib/plugin.py (hunk 1 above)
Comment (Paul-VW) [reply plugin.video.funcarrun/resources/lib/plugin.py]: fixed

File: plugin.video.funcarrun/resources/lib/plugin.py (hunk 3 above)
Comment (Paul-VW) [reply plugin.video.funcarrun/resources/lib/plugin.py]: fixed

File: plugin.video.funcarrun/resources/lib/plugin.py (hunk 4 above)
Comment (Paul-VW) [reply plugin.video.funcarrun/resources/lib/plugin.py]: fixed

File: plugin.video.funcarrun/resources/lib/plugin.py (hunk 2 above)
Comment (Paul-VW) [reply plugin.video.funcarrun/resources/lib/plugin.py]: fixed

**Review by enen92 — APPROVED** [review]

**Pull request merged**